src/main/java/com/aws/greengrass/mqttclient/spool/InMemorySpool.java

```diff
@@ -27,4 +29,9 @@ public void add(long id, SpoolMessage message) {
         messages.put(id, message);
     }
 
+    @Override
+    public List<Long> getAllSpoolMessageIds() {
+        return new ArrayList<>();
```

Comment:
Shouldn't it be returning `messages.keySet()` with `Set<Long>` return type?

Reply:
This interface method was written for the plugin to return the existing messageIds in the persistent database with order, which is why the return type must be List<Long> to keep the order. So that when the spooler starts the persistent queue can be synced with the queue in Spool.java.

Since InMemorySpool is not persistent, there will be no messageIds to sync when the spooler starts, so this implementation is just a placeholder that returns an empty list.

Reply:
This seems wrong. Irrespective of persistent or inMemory spool, what happens if higher level abstractions call this method, e.g. `CloudMessageSpool.getAllSpoolMessageIds()`

Reply:
Changing for next commit